nemo/src/io/formats/sparql.rs

```diff
@@ -0,0 +1,186 @@
+//! Handler for resources of type Sparql (java script object notation).
+
+use spargebra::Query;
+use std::sync::Arc;
+
+use nemo_physical::datavalues::{AnyDataValue, DataValue};
+use oxiri::Iri;
+
+//use reader::DsvReader;
+//use value_format::DsvValueFormats;
+//use writer::DsvWriter;
+
+use crate::{
+    io::format_builder::{
+        format_parameter, format_tag, value_type_matches, AnyImportExportBuilder, FormatParameter,
+        Parameters, StandardParameter,
+    },
+    rule_model::{
+        components::{import_export::Direction, term::value_type::ValueType},
+        error::validation_error::ValidationErrorKind,
+    },
+    syntax::import_export::{attribute, file_format},
+};
+
+use super::{ExportHandler, FormatBuilder, ImportHandler, ResourceSpec};
+
+use crate::io::formats::dsv::{value_format::DsvValueFormats, DsvHandler};
+
+format_tag! {
+    pub(crate) enum SparqlTag(SupportedFormatTag::Sparql) {
+        Sparql => file_format::SPARQL,
+    }
+}
+
+format_parameter! {
+    pub(crate) enum SparqlParameter(StandardParameter) {
+        Limit(name = attribute::LIMIT, supported_types = &[ValueType::Number]),
+        Format(name = attribute::FORMAT, supported_types = &[ValueType::Constant, ValueType::Tuple]),
+        Base(name = attribute::BASE, supported_types = &[ValueType::Constant]),
+        Endpoint(name = attribute::ENDPOINT, supported_types = &[ValueType::Constant]),
+        Query(name = attribute::QUERY, supported_types = &[ValueType::String]),
+        IgnoreHeaders(name = attribute::IGNORE_HEADERS, supported_types = &[ValueType::Boolean]),
+
+    }
+}
+
+impl FormatParameter<SparqlTag> for SparqlParameter {
+    fn required_for(&self, tag: SparqlTag) -> bool {
+        if matches!(tag, SparqlTag::Sparql) {
+            matches!(self, SparqlParameter::Endpoint) && matches!(self, SparqlParameter::Query)
```

Comment:
this needs to be `||`, not `&&`, otherwise it'll never be true. And then you can pull the disjunction into the pattern:
```rust
   matches!(self, Self::Endpoint | Self::Query)
```

Reply:
… and make it 
```rust
  matches!(tag, SparqlTag::Sparql) && matches!(self, Self::Endpoint | Self::Query)
```
to get rid of the `else { false }`